src/udiskslinuxpartition.c

```diff
@@ -311,7 +326,13 @@ handle_set_flags (UDisksPartition       *partition,
                                    bd_flags,
                                    &error))
         {
-          g_dbus_method_invocation_take_error (invocation, error);
+          g_dbus_method_invocation_return_error (invocation,
+                                                 UDISKS_ERROR,
+                                                 UDISKS_ERROR_FAILED,
+                                                 "Error setting partition flags on %s: %s",
+                                                 udisks_block_get_device (block),
+                                                 error->message);
+          udisks_simple_job_complete (UDISKS_SIMPLE_JOB (job), FALSE, error->message);
```

Comment:
``error`` needs to be ``g_clear_error()``-ed here or in the ``out:`` block. The same applies to almost all other modifications in this commit.